x-pack/plugins/security_solution/public/timelines/containers/index.test.tsx

```diff
@@ -0,0 +1,211 @@
+                });
+                return { unsubscribe: jest.fn() };
+              }),
+            };
+          }),
+        },
+      },
+      notifications: {
+        toasts: {
+          addWarning: jest.fn(),
+        },
+      },
+    },
+  }),
+}));
+
+const mockUseRouteSpy: jest.Mock = useRouteSpy as jest.Mock;
+jest.mock('../../common/utils/route/use_route_spy', () => ({
+  useRouteSpy: jest.fn(),
+}));
+
+mockUseRouteSpy.mockReturnValue([
+  {
+    pageName: SecurityPageName.overview,
+    detailName: undefined,
+    tabName: undefined,
+    search: '',
+    pathName: '/overview',
+  },
+]);
+
+describe('useTimelineEvents', () => {
+  beforeEach(() => {
+    mockSearch.mockReset();
+  });
+
+  const startDate: string = '2020-07-07T08:20:18.966Z';
+  const endDate: string = '3000-01-01T00:00:00.000Z';
+  const props: UseTimelineEventsProps = {
+    docValueFields: [],
+    endDate: '',
+    id: TimelineId.active,
+    indexNames: ['filebeat-*'],
+    isTimerangeInitializing: false,
+    fields: [],
+    filterQuery: '',
+    startDate: '',
+    limit: 25,
+    sort: initSortDefault,
+    skip: false,
+  };
+
+  test('init', async () => {
+    await act(async () => {
+      const { result, waitForNextUpdate } = renderHook<
+        UseTimelineEventsProps,
+        [boolean, TimelineArgs]
+      >((args) => useTimelineEvents(args), {
+        initialProps: { ...props },
+      });
+
+      // useEffect on params request
+      await waitForNextUpdate();
+      expect(result.current).toEqual([
+        false,
+        {
+          events: [],
+          id: TimelineId.active,
+          inspect: result.current[1].inspect,
+          loadPage: result.current[1].loadPage,
+          pageInfo: result.current[1].pageInfo,
+          refetch: result.current[1].refetch,
+          totalCount: -1,
+          updatedAt: 0,
+        },
+      ]);
+    });
+  });
+
+  test('happy path query', async () => {
+    await act(async () => {
+      const { result, waitForNextUpdate, rerender } = renderHook<
+        UseTimelineEventsProps,
+        [boolean, TimelineArgs]
+      >((args) => useTimelineEvents(args), {
+        initialProps: { ...props },
+      });
+
+      // useEffect on params request
+      await waitForNextUpdate();
+      rerender({ ...props, startDate, endDate });
+      // useEffect on params request
+      await waitForNextUpdate();
+
+      expect(mockSearch).toHaveBeenCalledTimes(1);
+      expect(result.current).toEqual([
+        false,
+        {
+          events: mockEvents,
+          id: TimelineId.active,
+          inspect: result.current[1].inspect,
+          loadPage: result.current[1].loadPage,
+          pageInfo: result.current[1].pageInfo,
+          refetch: result.current[1].refetch,
+          totalCount: 31,
+          updatedAt: result.current[1].updatedAt,
+        },
+      ]);
+    });
+  });
+
+  test('Mock cache for active timeline when switching page', async () => {
```

Comment:
Thanks for adding this test! The caching is working correctly for timeline itself (`TimelineId.active`), but there's an unexpected side effect that's causes timeline-based views to fetch on initial load page, but not when tabs are changed. To reproduce:

1. Open a new tab, and load Kibana on an **non**-Security solution route (e.g. the Kibana home page): http://localhost:5601/app/home#/
2. Navigate **directly** from the **non**-Security solution route to the `Detections` page via the Kibana menu (`Security > Detections`)

**Expected result**
- Alerts are displayed in the `Detections` page's timeline-based view (i.e. `Showing 59,800 alerts`)

3. Navigate to the `Overview` tab by clicking the `Overview` link

4. Navigate back to the `Detections` page by clicking the `Detections` link

**Expected result**
- The Alerts are once again displayed in the `Detections` page's timeline-based view (i.e. `Showing 59,800 alerts`)

**Actual result**

- The Alerts that were initially displayed in the `Detections` page's timeline-based view are not displayed (`Showing 0 alerts`)